src/PackageCompiler.jl

```diff
@@ -686,7 +686,17 @@ function create_sysimg_from_object_file(object_files::Vector{String},
     end
     mkpath(dirname(sysimage_path))
     # Prevent compiler from stripping all symbols from the shared lib.
-    o_file_flags = Sys.isapple() ? `-Wl,-all_load $object_files` : `-Wl,--whole-archive $object_files -Wl,--no-whole-archive`
+    if Sys.isapple()
+        cltools_version_cmd = `pkgutil --pkg-info=com.apple.pkg.CLTools_Executables`
+        cltools_version = match(r"version: (.*)\n", readchomp(cltools_version_cmd))[1]
+        if startswith(cltools_version, "15")
```

Comment:
I am wondering if we should make this >= 15 ?
I presume this issue will still be around in xcode 16 and we don't want to go back to the flags for <=14?

Reply:
I'm mildly torn on this. As-is, it forces us to revisit this fix for xcode 16 and investigate whether the issue has been solved or if our mildly hacky solution is still needed. Ultimately, though, that comes at the expense of (most likely) breaking development for anyone who upgrades to Xcode 16. I see no reason to assume that this will not continue to be an issue in xcode 16+, so making it `≥ 15` seems like the best course of action. Thanks!